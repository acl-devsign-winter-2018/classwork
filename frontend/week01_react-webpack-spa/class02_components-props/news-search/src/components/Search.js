import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Search extends Component {

  static propTypes = {
    onSearch: PropTypes.func.isRequired
  };

  state = {
    search: '',
    name: ''
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.onSearch(this.state);
  };

  handleChange = ({ target }) => {
    const { name, value } = target;
    this.setState({ [name]: value });
  };

  render() {
    const { category, name } = this.state;

    return (
      <form onSubmit={this.handleSubmit}>
        <input name="category" value={category} onChange={this.handleChange}/>
        <input name="name" value={name} onChange={this.handleChange}/>
        <button>Search</button>
        <pre>{JSON.stringify(this.state, true, 2)}</pre>
      </form>
    );
  }
}