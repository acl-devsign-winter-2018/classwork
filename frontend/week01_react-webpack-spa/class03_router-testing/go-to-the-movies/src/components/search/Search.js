import React, { Component } from 'react';
import Movies from '../movies/Movies';
import SearchForm from '../search/SearchForm';
import { search } from '../../services/movieApi';

export default class Search extends Component {
  
  state = {
    movies: null,
    error: null
  };

  handleSearch = searchTerm => {
    this.setState({ error: null });

    search(searchTerm)
      .then(({ Search }) => {
        this.setState({ movies: Search });
      })
      .catch(error => {
        this.setState({ error });
      });
  };
  
  render() {
    const { movies, error } = this.state;

    return (
      <div>
        <SearchForm onSearch={this.handleSearch}/>
        {error && <div>{error}</div>}
        {(!error && movies) && <Movies movies={movies}/>}
      </div>
    );
  }
}