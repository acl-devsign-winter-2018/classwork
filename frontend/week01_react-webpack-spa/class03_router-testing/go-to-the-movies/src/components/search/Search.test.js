import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Search from './Search';

describe('Search', () => {

  it('Renders as Design', () => {
    const wrapper = shallow(<Search onSearch={() => {}}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Calls onSearch with criteria entered', () => {
    let filter;
    const handleSearch = _filter => filter = _filter;

    const wrapper = mount(<Search onSearch={handleSearch}/>);

    const search = 'Star Wars';
    
    // input and click
    const input = wrapper.find('input');
    input.instance().value = search;
    const form = wrapper.find('form').instance();
    const button = wrapper.find('button');
    button.simulate('submit', { target: form });

    // did the handleSearch get called with the simulated input?
    expect(filter).toBe(search);
  });

});