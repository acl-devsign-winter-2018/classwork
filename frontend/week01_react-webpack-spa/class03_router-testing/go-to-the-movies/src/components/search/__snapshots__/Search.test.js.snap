// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Renders as Design 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <label>
      Search for Movies: 
      <input
        name="search"
        placeholder="enter movie search"
      />
    </label>
    <label>
      Â 
      <button>
        Search
      </button>
    </label>
  </fieldset>
</form>
`;
